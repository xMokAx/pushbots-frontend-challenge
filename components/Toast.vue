<template>
  <v-snackbar
    v-model="isToastShown"
    :timeout="3000"
    :color="toastOptions.color"
  >
    <v-icon v-if="toastOptions.icon" color="white" class="mr-2">{{
      toastOptions.icon
    }}</v-icon>
    <span>{{ toastOptions.text }}</span>
    <v-btn icon @click="CLOSE_TOAST"><v-icon>mdi-close-circle</v-icon></v-btn>
  </v-snackbar>
</template>

<script>
import { mapGetters, mapMutations } from 'vuex'
import { CLOSE_TOAST } from '../store/toast/mutation-types'
export default {
  computed: {
    isToastShown: {
      get() {
        return this.$store.state.toast.isToastShown
      },
      set() {
        this[CLOSE_TOAST]()
      }
    },
    ...mapGetters('toast', ['toastOptions'])
  },
  methods: mapMutations('toast', [CLOSE_TOAST])
}
</script>
