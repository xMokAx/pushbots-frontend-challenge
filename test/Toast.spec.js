import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount } from '@vue/test-utils'
import Toast from '../components/Toast'

Vue.use(Vuetify)

const computed = {
  isToastShown: {
    get() {
      return true
    },
    set() {
      return false
    }
  },
  toastOptions() {
    return {
      color: 'error',
      text: 'Email or Password is wrong',
      icon: 'mdi-alert'
    }
  }
}

describe('Toast', () => {
  it('Should match snapshot', () => {
    const wrapper = mount(Toast, {
      computed
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
