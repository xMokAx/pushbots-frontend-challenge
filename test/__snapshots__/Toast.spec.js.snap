// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Should match snapshot 1`] = `"<div class=\\"v-snack v-snack--active v-snack--bottom\\" name=\\"v-snack-transition\\"><div class=\\"v-snack__wrapper error\\"><div class=\\"v-snack__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-2 mdi mdi-alert theme--light white--text\\"></i> <span>Email or Password is wrong</span> <button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-close-circle theme--light\\"></i></span></button></div></div></div>"`;
